pkg/services/baremetal/host/host.go

```diff
@@ -101,15 +101,15 @@ func (s *Service) Reconcile(ctx context.Context) (result reconcile.Result, err e
 
 		// save host if it changed during reconciliation
 		if !reflect.DeepEqual(oldHost, *s.scope.HetznerBareMetalHost) {
-			result, err = SaveHostAndReturn(ctx, s.scope.Client, s.scope.HetznerBareMetalHost)
+			result, err = SaveHostAndReturn(ctx, s.scope.Client, s.scope.HetznerBareMetalHost) // the original result gets overwritten.
 		}
 	}()
 
 	// reconcile state
 	actResult := hostStateMachine.ReconcileState()
 	result, err = actResult.Result()
 	if err != nil {
-		return reconcile.Result{Requeue: true}, fmt.Errorf("action %q failed: %w", initialState, err)
+		return reconcile.Result{RequeueAfter: time.Second * 15}, fmt.Errorf("action %q failed: %w", initialState, err)
```

Comment:
@janiskemper  I added a delay here. Is this ok?

Reply:
is there a reason? I don't think this is good. Probably this will also break our unit tests

Reply:
apparently it doesn't, but it might as soon as we actually use this error for something... Because our timeout will be smaller than 15 seconds.

Reply:
ok, I reverted this